org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/GradleDistribution.java

```diff
@@ -8,57 +8,22 @@
 
 package org.eclipse.buildship.core;
 
-import java.io.File;
-import java.net.URI;
-
 /**
  * Represents a Gradle distribution.
+ * <p>
+ * Currently four different Gradle distribution types are supported.
+ * <ul>
+ * <li>{@link WrapperGradleDistribution}</li>
+ * <li>{@link LocalGradleDistribution}</li>
+ * <li>{@link RemoteGradleDistribution}</li>
+ * <li>{@link FixedVersionGradleDistribution}</li>
+ * </ul>
+ * New instances can be created with the factory methods in {@link GradleDistributions}.
  *
  * @author Donat Csikos
  * @since 3.0
  * @noimplement this interface is not intended to be implemented by clients
  */
-public abstract class GradleDistribution {
-
-    /**
-     * Creates a reference to a local Gradle installation.
-     *
-     * @param installationDir the local Gradle installation directory to use
-     * @return a new distribution instance
-     */
-    public static GradleDistribution forLocalInstallation(File installationDir) {
-        return org.eclipse.buildship.core.internal.util.gradle.GradleDistribution.forLocalInstallation(installationDir);
-    }
-
-    /**
-     * Creates a reference to a remote Gradle distribution. The appropriate distribution is
-     * downloaded and installed into the user's Gradle home directory.
-     *
-     * @param distributionUri the remote Gradle distribution location to use
-     * @return a new distribution instance
-     */
-    public static GradleDistribution forRemoteDistribution(URI distributionUri) {
-        return org.eclipse.buildship.core.internal.util.gradle.GradleDistribution.forRemoteDistribution(distributionUri);
-    }
-
-    /**
-     * Creates a reference to a specific version of Gradle. The appropriate distribution is
-     * downloaded and installed into the user's Gradle home directory.
-     *
-     * @param version the Gradle version to use
-     * @return a new distribution instance
-     */
-    public static GradleDistribution forVersion(String version) {
-        return org.eclipse.buildship.core.internal.util.gradle.GradleDistribution.forVersion(version);
-    }
+public interface GradleDistribution {
```

Comment:
❌ Should be an abstract class with 4 nested concrete classes, a.k.a a sealed class hierarchy.

Reply:
Fixed.